[tiognjjyi] iP

tiongjjyi opened the pull request:
## DukePro
> "Your mind is for having ideas, not holding them." -David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

DukePro frees your mind of having to remember things you need to do. It's,
-  text-based
-  easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,
1. download it from [here](https://github.com/tiongjjyi/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 💯 

Features:
- [ ] Managing tasks
- [ ] Managing deadline
- [ ] Reminders
 
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
        new Duke().run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tiongjjyi committed:
Level-0

tiongjjyi committed:
Level-1

tiongjjyi committed:
Level-1

tiongjjyi committed:
Revert "Level-1"

This reverts commit 54cdddf0b7d24046f289fca404e581b27f3ec1a5.

tiongjjyi committed:
Level-2

tiongjjyi committed:
Level-3

tiongjjyi committed:
Level-3

tiongjjyi committed:
Level-4

tiongjjyi committed:
Level-5

tiongjjyi committed:
Level-6

tiongjjyi committed:
A-TextUiTesting

tiongjjyi committed:
Level-7

tiongjjyi committed:
Level-7 merging

tiongjjyi committed:
Level-8

tiongjjyi committed:
Merge branch 'branch-level-8'

tiongjjyi committed:
A-MoreOOP

tiongjjyi committed:
A-Packages

tiongjjyi committed:
Modify Storage to load data

tiongjjyi committed:
Merge branch 'add-gradle-support'

tiongjjyi committed:
A-Gradle

tiongjjyi committed:
add testcases

tiongjjyi committed:
A-Jar

tiongjjyi committed:
A-JavaDoc

tiongjjyi committed:
A-CodingStandard

tiongjjyi committed:
Level-9

tiongjjyi committed:
Merge branch 'branch-Level-9'
Add find functionality.

tiongjjyi committed:
Merge branch 'branch-A-CodingStandard'
Do checkstyles.

tiongjjyi committed:
merge A-JavaDoc

KamJiaYue commented on src/main/java/duke/taskmanagement/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+package duke.taskmanagement;
+import java.time.LocalDate;
+public class Deadline extends Task {
```
Perhaps adding some space between these two lines would increase readability and neatness?

KamJiaYue commented on src/main/java/duke/taskmanagement/Storage.java:
```diff
@@ -0,0 +1,207 @@
+public class Storage {
+
+    private String filePath;
+    private List<Task> ls = new ArrayList<>();
```
Perhaps using taskList instead of ls would increase clarity of code?

KamJiaYue commented on src/main/java/duke/taskmanagement/Storage.java:
```diff
@@ -0,0 +1,207 @@
+                ls.add(task);
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```
I like how clever catching Errors just by one catch.

KamJiaYue commented on src/main/java/duke/taskmanagement/Storage.java:
```diff
@@ -0,0 +1,207 @@
+     *              change the status to done.
+     */
+    public void changeToDone(int index) {
+                List<String> lines = new ArrayList<>();
```
Any reason for the wide indents you used?

NgChunMan replied:
> Any reason for the wide indents you used?

Should this line require wide indents?

KamJiaYue commented on src/main/java/duke/taskmanagement/ToDo.java:
```diff
@@ -0,0 +1,42 @@
+     * @return A string that represent of status of the task, [X] if done,
+     * [ ] if not done.
+     */
+    @Override
+
+    public String toString(){
```
Perhaps the gap between @Override and the function might cause confusion?

s-peiran replied:
line 21: public String toString() {

tiongjjyi committed:
minor modification before Level-10

NgChunMan commented on src/main/java/duke/taskmanagement/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+        this.by = by;
+    }
+    /**
+     * Return the status icon of whether the task is done or nor
```
Perhaps an empty line between description and parameter section would be easier for reader to read?

NgChunMan commented on src/main/java/duke/taskmanagement/DukeException.java:
```diff
@@ -0,0 +1,45 @@
+    protected String typeOfTask;
+
+    /**
+     * Create new duke exception to handle
```
Perhaps the first sentence in the header comment starts with Creates more appropriate?

NgChunMan commented on src/main/java/duke/taskmanagement/DukeException.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    /**
+     * Return the String that shows the error
```
Perhaps the first sentence in the header comment starts with Returns and an empty line between description and parameter section is more appropriate?

NgChunMan commented on src/main/java/duke/taskmanagement/DukeException.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    /**
+     * Return the String that shows the error
```
... I noticed the same issue in several other places too.

NgChunMan commented on src/main/java/duke/taskmanagement/Event.java:
```diff
@@ -0,0 +1,39 @@
+        this.till = till;
+    }
+    /**
+     * Return the status icon of whether the task is done or nor
```
... I noticed the same issue in several other places too.

NgChunMan commented on src/main/java/duke/taskmanagement/Parser.java:
```diff
@@ -0,0 +1,101 @@
+    }
+
+    /**
+     * To read the command from user input and
```
... I noticed the same issue in several other places too

NgChunMan commented on src/main/java/duke/taskmanagement/Storage.java:
```diff
@@ -0,0 +1,207 @@
+            }
+
+    /**
+     * To modify the text file as the user modify the
```
... I noticed the same issue in several other places too.

NgChunMan commented on src/main/java/duke/taskmanagement/Duke.java:
```diff
@@ -0,0 +1,28 @@
+package duke.taskmanagement;
```
Good job in organising all the classes into the same package.

NgChunMan commented on src/main/java/duke/taskmanagement/TaskList.java:
```diff
@@ -0,0 +1,128 @@
+     *              text file and list.
+     */
+    public void deleteTask(int index) {
+        Task task = ls.get(index - 1);
```
Perhaps a more intuitive variable name here? Such as deletedTask?

NgChunMan commented on src/main/java/duke/taskmanagement/TaskList.java:
```diff
@@ -0,0 +1,128 @@
+     * @param index The index of task to mark as done.
+     */
+    public void mark(int index) {
+        Task task = ls.get(index-1);
```
Perhaps taskToBeMarked is more intuitive?

NgChunMan commented on src/main/java/duke/taskmanagement/TaskList.java:
```diff
@@ -0,0 +1,128 @@
+    public List<Task> find(String str) {
+        List<Task> tmpList = new ArrayList<>();
+        tmpList = ls;
+        tmpList = tmpList.stream().filter(string -> string.contains(str)).collect(Collectors.toList());
```
I like the way you make your List into a stream and find the task by using filter and collect methods in Java Stream class.

NgChunMan commented on src/main/java/duke/taskmanagement/Ui.java:
```diff
@@ -0,0 +1,102 @@
+
+    public void printFilterList(List<Task> filteredList) {
+        System.out.println("Here are the matching tasks in your list: ");
+        for(int i = 0; i < filteredList.size() ;i++) {
```
Perhaps a white space after the for keyword is better?
Perhaps semicolons in for statements followed by a space character are better?

NgChunMan commented on src/main/java/duke/taskmanagement/Ui.java:
```diff
@@ -0,0 +1,102 @@
+     */
+    public void printList(int size, List<Task> ls){
+        System.out.println("Here are the tasks in your list: ");
+        for(int i = 0; i < size;i++) {
```
Perhaps a white space after the for keyword is better?
Perhaps semicolons in for statements followed by a space character are better?

NgChunMan commented on src/main/java/duke/taskmanagement/Ui.java:
```diff
@@ -0,0 +1,102 @@
+        System.out.println("Got it. I've added this task:\n" + str + "\n" + "Now you have " + size + " tasks in the list.\n");
+    }
+
+    public void printFilterList(List<Task> filteredList) {
```
Is there a missing javadoc comment for this printFilterList method?

NgChunMan commented on src/main/java/duke/taskmanagement/TaskList.java:
```diff
@@ -0,0 +1,128 @@
+        storage.changeToDone(index);
+    }
+
+    public List<Task> find(String str) {
```
Is there a missing javadoc comment for this find method?

NgChunMan reviewed (commented):
LGTM. I noted a few minor coding standard violations. Just a few nits to fix.

tiongjjyi committed:
add GUI (incomplete)

s-peiran commented on src/main/java/duke/taskmanagement/Event.java:
```diff
@@ -0,0 +1,39 @@
+     * @return "1" if the task is done and "0" if the task is not done.
+     */
+    public String convertIsDone() {
+        return super.isDone ? "1" : "0";
```
consider using enumerations

s-peiran commented on src/main/java/duke/taskmanagement/Parser.java:
```diff
@@ -0,0 +1,101 @@
+            } else if (cmd.contains("deadline")) {
+                String[] parts = cmd.split("/by");
+                if (parts.length == 2) {
+                    String description = parts[0].replace("deadline", "").trim(); // Remove "deadline"
```
recommended to abstract the deadline details in the deadline class constructor.

s-peiran commented on src/main/java/duke/taskmanagement/Parser.java:
```diff
@@ -0,0 +1,101 @@
+                    Task task = new Deadline(description, formattedDate, false);
+                    ls.addDeadlineTask(task);
+                } else {
+                    DukeException exp = new DukeException("deadline");
```
exception should be try-catch -ed. 
i.e method should throw DukeException

s-peiran commented on src/main/java/duke/taskmanagement/Storage.java:
```diff
@@ -0,0 +1,207 @@
+     * @param lineIndexToDelete The index of the line (task) you
+     *                          want to delete.
+     */
+    public void deleteLine(int lineIndexToDelete) {
```
How can you write the method to make it less nested?

s-peiran reviewed (commented):
Good JavaDoc comments and very neat methods. Do consider some suggestions below

tiongjjyi committed:
complete GUI

tiongjjyi committed:
Minor modification before proceed to next increment

tiongjjyi committed:
Add assertion function

The current situation in the codebase exposes vulnerabilities due to insufficient input validation. This needs to change to enhance the overall robustness of the application.

In this commit, an assertion function is introduced to thoroughly validate user inputs before processing. This proactive approach ensures that only valid data is processed, reducing the risk of errors and security breaches.

This approach is chosen because it provides a centralized and consistent way to handle input validation, making the codebase more maintainable and secure.

tiongjjyi committed:
Merge pull request #1 from tiongjjyi/branch-A-Assertions

Add assertion function

tiongjjyi committed:
Follow coding style and format guidelines

The current situation in the codebase exhibits inconsistencies in coding style and formatting, which can lead to reduced code quality and maintainability. This needs to change to ensure a more consistent and readable codebase.

In this commit, we enforce coding style and format guidelines across the entire codebase. This includes addressing indentation, naming conventions, and code structure to adhere to established best practices.

This approach is being taken to promote a more uniform and maintainable codebase, making it easier for developers to collaborate and understand the code.

tiongjjyi committed:
Merge pull request #2 from tiongjjyi/branch-A-CodeQuality

Follow coding style and format guidelines

tiongjjyi committed:
Introduce 'help' command to provide user guidance -- Enhance user experience

The current situation in the application lacks a user-friendly way to assist users in understanding and utilizing available commands. This needs to change to improve the overall user experience.

In this commit, a 'help' command is implemented, allowing users to input 'help' to receive a list of available commands along with brief descriptions. This feature enhances user understanding and interaction with the application.

This approach is chosen to make the application more intuitive and user-friendly, ensuring that users can easily access information about available commands and their usage.

tiongjjyi committed:
Minor edits to this branch

tiongjjyi committed:
Add product screenshots to docs

tiongjjyi committed:
Add User Guide to README

tiongjjyi committed:
Modify some codes to follow SLAP

tiongjjyi committed:
Handle more invalid inputs from user

tiongjjyi committed:
Update features in User Guide

tiongjjyi committed:
Code clean up

tiongjjyi committed:
Handle invalid date format or empty date in Deadline task

tiongjjyi committed:
Remove unused method in Parser class

tiongjjyi committed:
Close window after bye command

tiongjjyi committed:
Give credit for reused work

tiongjjyi committed:
Complete javadocs

tiongjjyi committed:
Minor modification in help() method in Ui class

tiongjjyi committed:
Resolve JavaFX runtime components are missing on Windows issue
